src/ray/core_worker/core_worker.h

```diff
@@ -1849,6 +1857,20 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   uint32_t pid_;
 
   absl::flat_hash_set<ObjectID> deleted_generator_ids_;
+
+  /// Serialized runtime info env are cached.
+  /// TODO(hjiang):
+  /// 1. Better to use hash value for serialized runtime (key), or cap a max serialized
+  /// string length to avoid too much memory consumption.
+  /// 2. Implement a LRU cache, to cap max number of key-value pairs to limit max memory
```

Comment:
This is needed to merge this PR otherwise we will have unbounded memory usage.

Reply:
OK, let me make another PR for LRU cache under `util` folder

Reply:
I found there's an existing implementation in boost, updated implementation.